src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,61 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int INITIAL_SIZE = 10;
+    private static final double GROWTH_FACTOR = 1.5;
+    private int currentSize = INITIAL_SIZE;
+    private int size = 0;
+    private T[] table = createArray(currentSize);
 
+    private T[] createArray(int size) {
+        return (T[]) new Object[size];
+    }
+
+    public void push(T value) {
+        table = grow();
+        table[size] = value;
+        size++;
     }
 
     public T peek() {
-        return null;
+        if (size == 0) {
+            throw new EmptyStackException();
+        }
+        return table[size - 1];
     }
 
     public T pop() {
-        return null;
+        if (size == 0) {
+            throw new EmptyStackException();
+        }
+        T value = peek();
+        table[size - 1] = null;
+        size--;
+        return value;
     }
 
     public int size() {
-        return 0;
+        return size;
+    }
+
+    private T[] grow() {
+        if (size < currentSize) {
+            return table;
+        }
+        int newLength = (int) (table.length * GROWTH_FACTOR);
+        currentSize = (int) newLength;
+        T[] newTable = createArray(currentSize);
+        System.arraycopy(table,0,newTable,0,table.length);
+        return newTable;
+    }
```

Comment:
This could also be simplified like this for example (just do necessary checks in the `push` method before calling this one:

```java
        private void resize() {
        currentSize = (int) (currentSize * DEFAULT_MULTIPLAYER);
        System.arraycopy(stack,0,stack,0, currentSize);
    }
```

Reply:
ok, done